### PR TITLE
Players have ready state

### PR DESCRIPTION
**Pull request by gemcfadyen**
@jsuchy @ecomba This PR introduces an `is_ready` method so that the web players (a new type of player for the web game - a separate pr is being raised in the web package for this), can break out of the game loop after each move is taken.

### CODE REVIEW

File: lib/ai_player.rb [hunk 1]
```diff
@@ -12,6 +12,10 @@ def choose_move(board)
     minimax(board, true, board.vacant_indices.size, ALPHA, BETA).first[1]
   end
 
+  def is_ready?
```
Comment (jsuchy) [inline lib/ai_player.rb]: just `ready?`, is duplicates the predicate (? suffix).

File: lib/game.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
+require 'pry'
```
Comment (jsuchy) [inline lib/game.rb]: no no no no no :)

File: lib/game.rb [hunk 3]
```diff
@@ -18,7 +20,11 @@ def play
 
   attr_reader :board, :players
 
-  def game_in_progress
-    board.free_spaces? && !board.winning_combination?
+  def game_in_progress?(player)
```
Comment (jsuchy) [inline lib/game.rb]: maybe just ask for the current_player instead of passing it in.

File: lib/ai_player.rb (hunk 1 above)
Comment (ecomba) [inline lib/ai_player.rb]: :+1: agree with @jsuchy

File: lib/game.rb (hunk 2 above)
Comment (ecomba) [inline lib/game.rb]: :name_badge: ~~require pry~~

**Pull request merged**